### PR TITLE
Revision of Coding Standards

### PR DESCRIPTION
**Pull request by kschiess**

### CODE REVIEW

File: source/code-general.md [hunk 1]
```diff
@@ -49,37 +44,39 @@ All implementation code that can reasonably be covered by automated tests should
...
 ## Source control (i.e. Git) and versioning
 
-- One change per commit
 - Clear and explanatory commit messages, always mentioning the related feature or issue  if applicable (use GitHub commit messages integration, e.g. `fixes #321`)
+- Follow the format outlined in this [guide](https://chris.beams.io/posts/git-commit/).
+
+## Release Management
+
+We 
+
```
Comment (sgoumaz) [inline source/code-general.md]: If you want SC and versioning into separate sections I guess the former's title should be amended, but I'm not sure about the purpose of this change

File: README.md [hunk 2]
```diff
@@ -2,7 +2,6 @@
 
 Source content for our team guidelines, published on http://pryv.github.io/guidelines.
 
-
 ## Building
```
Comment (sgoumaz) [inline README.md]: We more or less used 2 spaces to separate document sections for readability when quickly skimming through a doc, as Markdown is full of blank lines, but I do find it's not needed if your MD editor makes headings very obvious.

File: source/code-general.md [hunk 3]
```diff
@@ -49,37 +44,39 @@ All implementation code that can reasonably be covered by automated tests should
 
 ### Code layout / style
 
-- Maximum line length: 100 characters
+- Maximum line length: 80 characters
```
Comment (sgoumaz) [inline source/code-general.md]: Why? I always found 80 chars an enemy of explicitly named identifiers

File: source/code-javascript.md [hunk 4]
```diff
@@ -12,50 +12,38 @@
...
 ### Miscellaneous
 
 - Always use strict equality: `===` or `!==`
-- Use `.bind(this)` (or better, `thisArg` when available) instead of aliasing (`var self = this`)
+- Use `.bind(this)` (or better, `thisArg` when available) instead of aliasing (`var self = this`). Look at block function syntax to avoid rebinding `this`.
 - Keep functions short and operating within a single level of abstraction
 - Prefer duck-typing to `instanceof` type testing
-- Use [lodash](http://lodash.com) as a general-purpose utility library when core Javascript constructs are not sufficient (beware of over-using it especially on Node.js where there are no cross-browser compatibility issues)
-
+- Use [Ramda](http://ramdajs.com/).
```
Comment (sgoumaz) [inline source/code-javascript.md]: A priori I like that as it promotes functional programming, but then maybe other guidelines are needed to support that

File: source/code-general.md (hunk 3 above)
Comment (kschiess) [inline source/code-general.md]: The limit is there for two reasons: a) wide code is bad code (that I guess we agree on) and b) when working with the code through various tools, limiting line lengths prevents becoming confused with badly wrapped code. 

A value of 80 is better for the tools I use (at rather large font sizes, I admit!) - but only as a matter of taste in the reasoning of b). 

I have found that a stricter limit tends to get you in trouble faster - and that I consider that a good thing to be in in this case. That's my only non-taste argument here.

**Comment by kschiess** [comment]
@sgoumaz With regards to your general remark: The goal here is to set standards for new code, not for old code. We won't start a conversion project.

File: source/code-general.md (hunk 1 above)
Comment (kschiess) [inline source/code-general.md]: I feel that releases should be managed internally and will probably be subject of a document to that effect. After consideration, I would rather remove this section altogether.

File: source/code-javascript.md (hunk 4 above)
Comment (kschiess) [inline source/code-javascript.md]: My goal here is to reduce the guidelines as far as possible, not bloat them. I would like to reduce process in favour of culture. 

But I am open to suggestions as to what needs to be said about usage of Ramda. Have you made bad experiences with a certain pattern involving Ramda? Can we document and prevent this?

File: source/code-general.md (hunk 3 above)
Comment (sgoumaz) [inline source/code-general.md]: 80 chars does get me into trouble, but in my experience not the desirable kind, even with 2-char indents – as I suggested, like just calling a explicitly-named method on an explicitly-named object with a few explicitly-named arguments. Maybe it's a question of how you use white space/new lines. But I'd like to ignore your point B considering today's monitor resolutions and software tools. If those 80 chars are close to your heart I suggest a team vote.

File: source/code-general.md (hunk 1 above)
Comment (sgoumaz) [inline source/code-general.md]: I agree about putting release management elsewhere. But git tagging and use of semver I consider versioning basics, so I'm still unsure why we should remove them.

File: source/code-javascript.md (hunk 4 above)
Comment (sgoumaz) [inline source/code-javascript.md]: Agreed on process vs culture. That may not be something we've done very well till now. If you manage to get everyone pairing regularly I'll send you beer!

About Ramda I never used it, I just had a look. I meant that it looks good, but that mentioning the lib may not be enough to effectively support a more functional programming style (thus a better use of Ramda) in the team. So maybe more context (e.g. a discussion) or guidelines would be useful, as there's nothing related at the moment.

**Comment by sgoumaz** [comment]
@kschiess Yes I didn't understand you'd convert the existing code as a project, it's just that old code and new code will unavoidably coexist so there's a cost regardless. Even if just that of 'style switching' when devs jump between old and new. So basically I meant every change should be based on more than taste. But again, overall I'm extremely happy with what you've brought already.

**Pull request closed without merging**